### PR TITLE
Add interface for caching sorted affine peers

### PR DESCRIPTION
**Pull request by kriskowal**
r @raynos @jcorbin

### CODE REVIEW

File: egress-nodes.js [hunk 1]
```diff
@@ -78,6 +92,41 @@ EgressNodes.prototype.setKValueFor = function setKValueFor(serviceName, k) {
...
+    }
+    return affinePeers;
+};
+
+EgressNodes.prototype.computeAffinePeers = function computeAffinePeers(serviceName) {
+    var self = this;
+
+    assert(
+        self.ringpop !== null,
+        'EgressNodes#exitsFor cannot be called before EgressNodes has ' +
```
Comment (rf) [inline egress-nodes.js]: should be `EgressNodes#computeAffinePeers`

File: egress-nodes.js [hunk 2]
```diff
@@ -78,6 +92,41 @@ EgressNodes.prototype.setKValueFor = function setKValueFor(serviceName, k) {
...
+
+    // Unique peers are retained in sorted order
+    var peers = [];
+    var k = self.kValueFor(serviceName);
+    for (var i = 0; i < k; i++) {
+        var shardKey = serviceName + '~' + i;
+        var peer = self.ringpop.lookup(shardKey);
+        var j = sortedIndexOf(peers, peer);
+        if (j < 0) {
+            peers.splice(~j, 0, peer);
```
Comment (jcorbin) [inline egress-nodes.js]: this splice is so not like you... can't you just `peers.push(peer);`?

File: egress-nodes.js [hunk 3]
```diff
@@ -57,10 +59,22 @@ EgressNodes.prototype.setRingpop = function setRingpop(ringpop) {
 
     self.ringpop.on('membershipChanged', onMembershipChanged);
     function onMembershipChanged() {
-        self.membershipChangedEvent.emit(self);
+        self.onMembershipChanged();
     }
 };
 
+EgressNodes.prototype.onMembershipChanged = function onMembershipChanged() {
```
Comment (Raynos) [inline egress-nodes.js]: onRingChanged ?

**Comment by Raynos** [comment]
@jcorbin Did we do this in #111

**Comment by jcorbin** [comment]
Enough for now yes, if we ever want to revisit micro-ops like updating a sorted list incrementally, we can revisit.

**Pull request closed without merging**